src/Datadog.Trace.ClrProfiler.Managed/Integrations/Testing/AsyncTool.cs

```diff
@@ -0,0 +1,215 @@
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Datadog.Trace.ClrProfiler.Integrations.Testing
+{
+    /// <summary>
+    /// This class is similar to AsyncHelper but removes the params array allocation, the string key allocation and the invoke by reflection.
```

Comment:
Is the goal to replace (or merge) the current `AsyncHelper` with this later on?

Reply:
I think Kevin is working on a similar performance optimisation for the current AsyncHelper, he is aware of this implementation. The goal is once we have that merged, I'll remove this class and use that one.

Reply:
Thanks for clarifying.